src/components/cc-popover/cc-popover.js

```diff
@@ -0,0 +1,217 @@
+
+    /** @type {boolean} Whether the popover is opened */
+    this.isOpen = false;
+
+    /** @type {PopoverPosition} Sets the position of the popover relative to the `button` element. */
+    this.position = 'bottom-left';
+
+    /** @type {Ref<HTMLElement>} */
+    this._contentRef = createRef();
+
+    /** @type {Ref<HTMLElement>} */
+    this._buttonRef = createRef();
+
+    this._onOutsideClickHandler = new EventHandler(window, 'click', (event) => {
+      const contentElement = this._contentRef.value;
+      if (contentElement != null && !event.composedPath().includes(contentElement)) {
+        this.close();
+      }
+    });
+    this._onEscapeKeyHandler = new EventHandler(this, 'keydown', (event) => {
+      if (event.key === 'Escape') {
+        this.close();
+      }
+    });
+  }
+
+  /**
+   * Opens the popover.
+   */
+  open () {
+    if (!this.isOpen) {
+      this.isOpen = true;
+      dispatchCustomEvent(this, 'open');
+    }
+  }
+
+  /**
+   * Closes the popover.
+   */
+  close () {
+    if (this.isOpen) {
+      this.isOpen = false;
+      this._buttonRef.value?.focus();
+      dispatchCustomEvent(this, 'close');
+    }
+  }
+
+  /**
+   * Toggle the popover display.
+   */
+  toggle () {
+    if (this.isOpen) {
+      this.close();
+    }
+    else {
+      this.open();
+    }
+  }
+
+  updated (_changedProperties) {
+    if (_changedProperties.has('isOpen')) {
+      if (this.isOpen) {
+        this._onOutsideClickHandler.connect();
+        this._onEscapeKeyHandler.connect();
+      }
+      else {
+        this._onOutsideClickHandler.disconnect();
+        this._onEscapeKeyHandler.disconnect();
+      }
+    }
+  }
+
+  disconnectedCallback () {
+    super.disconnectedCallback();
+    this._onOutsideClickHandler.disconnect();
+    this._onEscapeKeyHandler.disconnect();
+  }
+
+  render () {
+    return html`
+      <cc-button
+        ${ref(this._buttonRef)}
+        aria-expanded=${this.isOpen ? 'true' : 'false'}
+        @cc-button:click=${this.toggle}
+        .accessibleName=${this.accessibleName}
+        ?hide-text=${this.hideText} 
+        .icon=${this.icon}
+      >
+        <slot name="button-text"></slot>
+      </cc-button>
+
+      ${this.isOpen ? html`
+        <div class="content ${this.position.replace('-', ' ')}" ${ref(this._contentRef)}>
+          <slot></slot>
+        </div>
+      ` : null}
+    `;
+  }
+
+  static get styles () {
+    return [
+      // language=CSS
+      css`
+        :host {
+          position: relative;
+          display: block;
+
+          --gap: 0.4em;
+        }
+        
+        .content {
+          position: absolute;
+          z-index: var(--cc-popover-z-index, 999);
+          padding: 0.5em;
+          border: 1px solid var(--cc-color-border-neutral, #aaa);
+          background-color: var(--cc-color-bg-default, #fff);
+          border-radius: 0.25em;
+          box-shadow: 0 2px 4px rgb(38 38 38 / 25%),
+            0 5px 15px rgb(38 38 38 / 25%);
```

Comment:
question: do we have tokens for this?

Reply:
we don't, it's in our list of things to tackle here: #398